pkg/composableschemadsl/compiler/translator.go

```diff
@@ -273,6 +261,64 @@ func translateObjectDefinition(tctx translationContext, defNode *dslNode) (*core
 	return ns, nil
 }
 
+// NOTE: this function and the translatePartial variation differ in how they treat a missing reference.
+// This function treats that as an error state, since all partials should be resolved by that point;
+// the partial variant treats that as a skip.
+func translateDefinitionRelationsAndPermissions(tctx translationContext, astNode *dslNode) ([]*core.Relation, error) {
```

Comment:
I imagine there's some deduplication to be done between this and the function below, but the point at which they differ is stuck in the middle. I'd be fine leaving this as a target for future refactoring.

Reply:
pass a flag and change behavior on that bool